File: ReadCsvFindMinMaxPrice/1.39vs0.17/ReadFileFindMinMaxPrice.php
<?php
echo 'Find min and max prices...'.PHP_EOL;

$fp = fopen('../test.csv', 'r');

$minProductName = '';
$minPrice = 4294967296;

$maxProductName = '';
$maxPrice = 0;

while (!feof($fp)) {
    $row = fgets($fp);
    $posDelimiter = strpos($row, ',');
    $price = (float) substr($row, $posDelimiter+1);

    if ($price <= 0) {
        continue;
    }
    if ($price < $minPrice) {
        $minPrice = $price;
        $minProductName = substr($row, 0, $posDelimiter);
    }
    if ($price > $maxPrice) {
        $maxPrice = $price;
        $maxProductName = substr($row, 0, $posDelimiter);
    }
}
fclose($fp);

print_r([$minProductName, $minPrice, $maxProductName, $maxPrice]);

File: ReadCsvFindMinMaxPrice/1.39vs0.17/ReadFileFindMinMaxPrice.pas
program ReadFileFindMinMaxPrice;

uses sysutils;

var f:text;
    s,minProductName,maxProductName:string;
    price, minPrice, maxPrice: real;
    posDelimiter: integer;
begin
writeln('Find min and max prices...');

minProductName := '';
minPrice := 4294967296;

maxProductName := '';
maxPrice := 0;

assign(f,'../test.csv');
reset(f);

while not eof(f) do
 begin
    readln(f,s);
    posDelimiter := pos(',',s);
    price := strtofloat(copy(s,posDelimiter+1));

    if (price <= 0) then continue;

    if (price < minPrice) then
        begin
        minPrice:=price;
        minProductName := copy(s,1,posDelimiter-1);
        end;
    if (price > maxPrice) then
        begin
        maxPrice:=price;
        maxProductName := copy(s,1,posDelimiter-1);
        end;
 end;
close(f);
writeln(minProductName,' ',minPrice);
writeln(maxProductName, ' ',maxPrice);
end.

